plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -204,16 +227,44 @@ def get_metadata(self):
         except ValueError, e:
             raise FailedException(str(e))
 
-        _logger.info(_('Downloading metadata files.'))
-
-        metadata_files.download_metadata_files()
-        self.downloader = None
-        _logger.info(_('Generating metadata databases.'))
-        metadata_files.generate_dbs()
-        self.import_unknown_metadata_files(metadata_files)
-        # TODO: verify metadata
-        #metadata_files.verify_metadata_files()
-        return metadata_files
+        scratchpad = self.sync_conduit.get_scratchpad() or {}
+        previous_revision = scratchpad.get(constants.REPOMD_REVISION_KEY, 0)
+        previous_skip_set = set(scratchpad.get(constants.PREVIOUS_SKIP_LIST, []))
```

Comment:
The idea here is that if a previous sync has the skip_list set to skip a type, and then the user removes that from the skip list and does a sync, we don't want to skip the sync just because the metadata didn't change.

So if anything was removed from the previous skip list, we will do a full sync.